src/Development/IDE/GHC/Compat.hs

```diff
@@ -182,3 +171,50 @@ dontWriteHieFiles d =
 #else
     d
 #endif
+
+nameListFromAvails :: [AvailInfo] -> [(SrcSpan, Name)]
+nameListFromAvails as =
+  map (\n -> (nameSrcSpan n, n)) (concatMap availNames as)
+
+#if !MIN_GHC_API_VERSION(8,8,0)
+-- Reimplementations of functions for HIE files for 8.6.5
```

Comment:
Can this comment be relaxed to include older versions of GHC?

Reply:
I only tested with 8.6.5 but I imagine it works for most recent version.

Reply:
I mean, will it work for GHC 8.4.x? ghcide still supports those versions, so if it doesn't work we would need to keep `supportsHieFiles` around

Reply:
It should work but I didn't test it.